.circleci/config.yml

```diff
@@ -0,0 +1,49 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/micrometer
+    environment:
+      GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+
+      - checkout
+
+      - restore_cache:
+          key: gradle-dependencies-{{ checksum "build.gradle" }}
+
+      - run:
+          name: downloadDependencies
+          command: ./gradlew downloadDependencies --console=plain
+
+      - save_cache:
+          key: gradle-dependencies-{{ checksum "build.gradle" }}
+          paths:
+            - ~/.gradle
+
+      - run:
+          name: run tests
+          command: ./gradlew test
+
+      - run:
+          name: collect test reports
+          when: always
+          command: |
+            mkdir -p ~/micrometer/test-results/junit/
+            find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} ~/micrometer/test-results/junit/ \;
+
+      - store_test_results:
+          path: ~/micrometer/test-results/
+
+      - store_artifacts:
+          path: build/libs
+
+      - deploy:
+          name: Deployment
+          command: |
+            if [ "$CIRCLE_BRANCH" = "master" ] || [ "$CIRCLE_BRANCH" = "1.0.x" ]; then
```

Comment:
This also doesn't prevent the publishing from happening on forks (pull requests) that use the branches `master` or `1.0.x`, which we should fix, but is also an existing issue.